antarest/study/service.py

```diff
@@ -2750,3 +2752,31 @@ def delete_file_or_folder(self, study_id: str, path: str, current_user: JWTUser)
         cache_id = f"{CacheConstants.RAW_STUDY}/{study.id}"
         updated_tree = study_tree.get()
         self.storage_service.get_storage(study).cache.put(cache_id, updated_tree)  # type: ignore
+
+    def create_folder(self, study_id: str, path: str, current_user: JWTUser) -> None:
+        study = self.get_study(study_id)
+        assert_permission(current_user, study, StudyPermissionType.WRITE)
+
+        url = [item for item in path.split("/") if item]
+        if len(url) < 2 or url[0] != "user":
+            raise FolderCreationNotAllowed(f"the given path isn't inside the 'User' folder: {path}")
+        if url[1] == "expansion":
+            raise FolderCreationNotAllowed(f"the given path shouldn't be inside the 'expansion' folder: {path}")
+
+        study_tree = self.storage_service.raw_study_service.get_raw(study, True).tree
```

Comment:
Don't we want to support this for variant studies too ?

If not, shouldn't we throw an explicit error if the study is not a raw study ?

If yes, we need to do the editing through a command (probably the existing UpdateConfig)

Reply:
I didn't think about this at all. I believe we have the same issue in the deletion endpoint just above. I'll test it.
The easy solution will be to forbid it but it doesn't really make sense from an user perspective so I'll think about something.

Reply:
Same, I don't think about this use case. I'm for the easy solution from Martin